cpp/openlocationcode.cc

```diff
@@ -55,11 +54,15 @@ double compute_precision_for_length(int code_length) {
   return pow_neg(internal::kEncodingBase, -3) / pow(5, code_length - 10);
 }
 
-// Finds the position of a char in the encoding alphabet.
+// Returns the position of a char in the encoding alphabet, or -1 if invalid.
 int get_alphabet_position(char c) {
-  const char* end = internal::kAlphabet + internal::kEncodingBase;
-  const char* match = std::find(internal::kAlphabet, end, c);
-  return (end == match)? -1 : (match - internal::kAlphabet);
+  // Lookup table of alphabet positions of characters 'C' through 'X'.
+  static const int kPositionLUT['X' - 'C' + 1] = { 8, -1, -1, 9, 10, 11, -1, 12,
+      -1, -1, 13, -1, -1, 14, 15, 16, -1, -1, -1, 17, 18, 19 };
```

Comment:
Any reason to put this inside the method as opposed to right below the definition of kAlphabet?

Reply:
I felt this piece of data was more specific to code in the function, so that's where I put it, but I could go either way.